client/client-multi/client-objective-c/Kaa/DefaultOperationDataProcessor.m

```diff
@@ -0,0 +1,269 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#import "DefaultOperationDataProcessor.h"
+#import "AvroBytesConverter.h"
+#import "KaaLogging.h"
+
+#define TAG @"DefaultOperationDataProcessor >>>"
+
+@interface DefaultOperationDataProcessor ()
+
+@property (nonatomic,strong) AvroBytesConverter *requestConverter;
+@property (nonatomic,strong) AvroBytesConverter *responseConverter;
+
+@property (atomic) int requestsCounter;
+
+@property (nonatomic,strong) id<MetaDataTransport> mdTransport;
+@property (nonatomic,strong) id<ConfigurationTransport> configurationTransport;
+@property (nonatomic,strong) id<EventTransport> eventTransport;
+@property (nonatomic,strong) id<NotificationTransport> notificationTransport;
+@property (nonatomic,strong) id<ProfileTransport> profileTransport;
+@property (nonatomic,strong) id<UserTransport> userTransport;
+@property (nonatomic,strong) id<RedirectionTransport> redirectionTransport;
+@property (nonatomic,strong) id<LogTransport> logTransport;
+
+@end
+
+@implementation DefaultOperationDataProcessor
+
+- (instancetype)init {
+    self = [super init];
+    if (self) {
+        self.requestConverter = [[AvroBytesConverter alloc] init];
+        self.responseConverter = [[AvroBytesConverter alloc] init];
+    }
+    return self;
+    
+}
+
+- (void)setRedirectionTransport:(id<RedirectionTransport>)transport {
+    @synchronized(self) {
+        _redirectionTransport = transport;
+    }
+}
+
+- (void)setMetaDataTransport:(id<MetaDataTransport>)transport {
+    @synchronized(self) {
+        self.mdTransport = transport;
+    }
+}
+
+- (void)setConfigurationTransport:(id<ConfigurationTransport>)transport {
+    @synchronized(self) {
+        _configurationTransport = transport;
+    }
+}
+
+- (void)setEventTransport:(id<EventTransport>)transport {
+    @synchronized(self) {
+        _eventTransport = transport;
+    }
+}
+
+- (void)setNotificationTransport:(id<NotificationTransport>)transport {
+    @synchronized(self) {
+        _notificationTransport = transport;
+    }
+}
+
+- (void)setProfileTransport:(id<ProfileTransport>)transport {
+    @synchronized(self) {
+        _profileTransport = transport;
+    }
+}
+
+- (void)setUserTransport:(id<UserTransport>)transport {
+    @synchronized(self) {
+        _userTransport = transport;
+    }
+}
+
+- (void)setLogTransport:(id<LogTransport>)transport {
+    @synchronized(self) {
+        _logTransport = transport;
+    }
+}
+
+- (void)processResponse:(NSData *)data {
+    @synchronized(self) {
+        if (!data) {
+            DDLogError(@"%@ Can't process nil response", TAG);
+            return;
+        }
+        
+        SyncResponse *syncResponse = (SyncResponse *)[self.responseConverter fromBytes:data object:[[SyncResponse alloc] init]];
```

Comment:
Need to check SyncResponse.status on SYNC_RESPONSE_RESULT_TYPE_PROFILE_RESYNC, if this command is received, do profile re-sync. See details https://github.com/kaaproject/kaa/blob/master/client/client-multi/client-cpp/impl/channel/SyncDataProcessor.cpp#L250